backend/core/src/main/java/org/sonarsource/sonarlint/core/tracking/SecurityHotspotMatchingService.java

```diff
@@ -201,7 +213,55 @@ private void updateStorageAndNotifyClient(String connectionId, SecurityHotspotCh
         hotspot.setAssignee(assignee);
       }
     });
-    client.didReceiveServerHotspotEvent(new DidReceiveServerHotspotEvent(connectionId, projectKey, event.getFilePath()));
+  }
+
+  private static Map<URI, List<RaisedHotspotDto>> updateHotspotsWithEventData(Map<URI, List<RaisedHotspotDto>> previouslyRaisedFindings, SecurityHotspotClosedEvent event) {
+    var hotspotKey = event.getHotspotKey();
+    return previouslyRaisedFindings.entrySet().stream()
+      .collect(toMap(Map.Entry::getKey, entry -> entry.getValue().stream()
+        .filter(it -> !hotspotKey.equals(it.getServerKey())).collect(Collectors.toList())));
+  }
+
+  private static Map<URI, List<RaisedHotspotDto>> updateHotspotsWithEventData(Map<URI, List<RaisedHotspotDto>> previouslyRaisedFindings, SecurityHotspotChangedEvent event) {
+    var hotspotKey = event.getHotspotKey();
+    return previouslyRaisedFindings.entrySet().stream()
+      .collect(toMap(Map.Entry::getKey, entry -> {
+        var unchangedHotspots = entry.getValue().stream()
```

Comment:
You end up adding changed hotspots to this list, so maybe we could find a more precise name

Reply:
True. Not sure the new name is much better, but I tried :)